### DISCUSSION TITLE
Website unusually takes 30s to load

### DISCUSSION THREAD

**Discussion by HeyMyian** [vkbo/novelWriter#1597] [Help] [answered]
The novelwriter.io website takes extraordinarily long to load. No other website has this problem for me, I'm on 100Mbps down. I've noticed this issue a couple of times in the recent past, so this is not just right now. Tested on the most recent versions of Firefox and Chrome.

With cache disabled, the time it takes to finish loading the site is ~30 seconds, tried several times. With cache enabled, it does load faster, although I can still watch the images being loaded in line by line as if it were 1995, which hints at the same issue.

Does anyone have the same problem? What's going on?

Here is a screenshot from the Firefox developer tools:

![2023-11-12___15_50_14](https://github.com/vkbo/novelWriter/assets/60368695/570e9d18-aae7-4d7c-a8cb-a3308bc6cfd3)

**Comment by HeyMyian** [comment]
The same site (homepage) with cache enabled.
Blue bar = waiting, green bar = receiving

![2023-11-12___16_16_22](https://github.com/vkbo/novelWriter/assets/60368695/46eb81ff-6fd8-4506-a514-b06b02567c10)

**Reply by vkbo** [reply]
Interesting. It loads fast for me. It's hosted with a fairly cheap host in Norway though. I've considered moving it to GitHub's pages solution instead. Since it is all static content, that's pretty much trivial.

For comparison: 
![image](https://github.com/vkbo/novelWriter/assets/1619840/1e848b11-5d79-4b4c-b8b8-78a5e5d70d18)

**Reply by vkbo** [reply]
I'll look into moving to GitHub some other time though, as the same day as I made a release is probably not the best time. It's been on my todo list for a while.

I've held off because I've considered adding a php script to handle help references from the application. Like if I wanted to add links from the app to certain help pages on the website, but make sure they have a key that doesn't depend on the file structure of the website, which sometimes changes. I can achieve the same with static redirects though.

**Reply by HeyMyian** [reply]
Thanks, it seems to not be an issue with the website software. I have a lot of packet loss to your hoster's IP (ran for about 200 packets, this is an excerpt):

```
Ping statistics for 194.63.249.218:
Packets: Sent = 39, Received = 31, Lost = 8 (20% loss), 
Approximate round trip times in milli-seconds: 
Minimum = 38ms, Maximum = 43ms, Average = 39ms     
```

Same for the IP your hoster's own domain points to. So it might be a local problem with my ISP (?) and I'll be patient for the move to GitHub.

**Reply by vkbo** [reply]
I mean, it's deployed to GitHub now, it just doesn't have the full domain URL: https://vkbo.github.io/novelWriter.io/

**Comment by johnblommers** [comment]
> The novelwriter.io website takes extraordinarily long to load. No other website has this problem for me,
> Does anyone have the same problem? What's going on?
> 

The website loads almost instantly here in the USA on my iPhone over Wi-Fi. Likely an issue with congestion or packet loss on your ISP connection.

**Comment by HeyMyian** [comment, marked as answer]
I just wanted to write this up in case anyone else comes across this for any reason.

This seems to be a "known" problem with my German ISP Telekom. The highlighted hop on 193.158.39.70 has caused various similar issues for other Telekom customers in the past. I suspect other German visitors to novelwriter.io might experience similar packet loss, but if you're going to point the domain to GitHub soon anyway it's going to be okay.

![Screenshot_2023-11-12_22-27-11](https://github.com/vkbo/novelWriter/assets/60368695/b369343c-9c81-400f-a392-d761c597c0c3)

The problem is only with IPv4. With IPv6 it's normal.
TIL about mtr. Cool tool :slightly_smiling_face:
